src/client/contexts/GoogleAnalyticsContext.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {
+  FC,
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from 'react'
+import { useLocation } from 'react-router-dom'
+import { GoogleAnalyticsService } from '../services'
+
+const {
+  GA_USER_EVENTS,
+  initializeGA,
+  setGAUserId,
+  sendPageView,
+  sendUserEvent,
+  sendTiming,
+  sendException,
+} = GoogleAnalyticsService
+
+interface GoogleAnalyticsContextProps {
+  GA_USER_EVENTS: {
+    SUBMIT: string
+  }
+  setGAUserId: (userId: number | null) => void
+  sendPageView: (path: string) => void
+  sendUserEvent: (action: string, label?: string, value?: number) => void
+  sendTiming: (category: string, variable: string, value: number) => void
+  sendException: (description: string) => void
+}
+
+export const GoogleAnalyticsContext = createContext<
+  GoogleAnalyticsContextProps | undefined
+>(undefined)
+
+export const useGoogleAnalytics = (): GoogleAnalyticsContextProps => {
+  const GoogleAnalytics = useContext(GoogleAnalyticsContext)
+  if (!GoogleAnalytics)
+    throw new Error(
+      'useGoogleAnalytics must be used within an GoogleAnalyticsProvider'
+    )
+  return GoogleAnalytics
+}
+
+export const GoogleAnalyticsProvider: FC = ({ children }) => {
+  const [isLoaded, setLoaded] = useState(false)
+
+  const location = useLocation()
+
+  useEffect(() => {
+    try {
+      initializeGA()
+    } catch (err) {
+      // TODO write error
```

Comment:
This is currently failing silently and swallowing the exception. If this is intentional, comment accordingly, otherwise, at the very least, log the exception, or let it pass through.